boards/FSM/Src/Io/Io_FlowMeters.c

```diff
@@ -0,0 +1,81 @@
+#include <assert.h>
+#include "main.h"
+#include "Io_SharedFreqOnlyPwmInput.h"
+#include "Io_FlowMeters.h"
+
+struct FlowMeters_Context
+{
+    struct FreqOnlyPwmInput *pwm_input;
+    TIM_HandleTypeDef *      htim;
+    HAL_TIM_ActiveChannel    active_channel;
+};
+
+static struct FlowMeters_Context primary_flow_meter, secondary_flow_meter;
+
+void Io_FlowMeters_Init(TIM_HandleTypeDef *htim)
+{
+    assert(htim != NULL);
+
+    primary_flow_meter.htim   = htim;
+    secondary_flow_meter.htim = htim;
+
+    primary_flow_meter.active_channel   = HAL_TIM_ACTIVE_CHANNEL_1;
+    secondary_flow_meter.active_channel = HAL_TIM_ACTIVE_CHANNEL_2;
+
+    primary_flow_meter.pwm_input = Io_SharedFreqOnlyPwmInput_Create(
+        htim, TIMx_FREQUENCY / TIMx_PRESCALER, TIM_CHANNEL_1,
+        TIMx_AUTO_RELOAD_REG);
+    secondary_flow_meter.pwm_input = Io_SharedFreqOnlyPwmInput_Create(
+        htim, TIMx_FREQUENCY / TIMx_PRESCALER, TIM_CHANNEL_2,
+        TIMx_AUTO_RELOAD_REG);
+}
+
+float Io_FlowMeters_GetPrimaryFlowRate(void)
+{
+    return Io_SharedFreqOnlyPwmInput_GetFrequency(
+               primary_flow_meter.pwm_input) /
+           7.5f;
+}
+
+float Io_FlowMeters_GetSecondaryFlowRate(void)
+{
+    return Io_SharedFreqOnlyPwmInput_GetFrequency(
+               secondary_flow_meter.pwm_input) /
+           7.5f;
+}
+
+void Io_FlowMeters_TickPrimary(void)
+{
+    Io_SharedFreqOnlyPwmInput_Tick(primary_flow_meter.pwm_input);
```

Comment:
This function isn't called anywhere else other than in this .c file. In fact, we can just write 

```
    if (htim == primary_flow_meter.htim &&
        htim->Channel == primary_flow_meter.active_channel)
    {
        Io_FlowMeters_TickPrimary();
    }
```
instead of 
```
    if (htim == primary_flow_meter.htim &&
        htim->Channel == primary_flow_meter.active_channel)
    {
        Io_SharedFreqOnlyPwmInput_Tick(primary_flow_meter.pwm_input);
    }
```

We don't want to create unncessary public functions

Reply:
I think you meant the reverse? I fixed it so hopefully that was what u meant

Reply:
Oh yeah i mixed it up, lgtm now